docs/Gopkg.toml.md

```diff
@@ -0,0 +1,104 @@
+# Gopkg.toml Documenation
+
+## `[metadata]`
+metadata can exist at the root of the manifest as well as under `constraint` and `override`. 
+At the root metadata defines metadata about the project that could be used by other independent systems. The metadata defined here will be ignored by dep.
+Under `constraint` or `override` metadata defines metadata about the dependency or override that could be used by other independent systems. The metadata defined here will be ignored by dep.
+```toml
+[metadata]
+key1 = "value that convey data to other systems"
+system1-data = "value that is used by a system"
+system2-data = "value that is used by another system"
+```
+
+## `required`
+required lists a set of packages (not projects) that must be included in Gopkg.lock. This list is merged with the set of packages imported by the current project. Use it when your project needs a package it doesn't explicitly import - including "main" packages.
+ ```toml
+required = ["github.com/user/thing/cmd/thing"]
+```
+
+## `ignored`
+ignored lists a set of packages (not projects) that are ignored when dep statically analyzes source code. Ignored packages can be in this project, or in a dependency.
+```toml
+ignored = ["github.com/user/project/badpkg"]
+```
```

Comment:
Let's move `required` and `ignored` above `metadata`. That's because placing `metadata` above `required` and `ignored` might result in people using the same order in Gopkg.toml file as well, which would become a malformed manifest.